p2p/peer.py

```diff
@@ -361,12 +364,13 @@ def close(self):
         # too much time is being spent on this again, we need to consider running this in a
         # ProcessPoolExecutor(). Need to make sure we don't use all CPUs in the machine for that,
         # though, otherwise asyncio's event loop can't run and we can't keep up with other peers.
-        decoded_msg = cmd.decode(msg)
+        decoded_msg = cast(Dict[str, Any], cmd.decode(msg))
         self.logger.debug("Successfully decoded %s msg: %s", cmd, decoded_msg)
         return cmd, decoded_msg
 
     def handle_p2p_msg(self, cmd: protocol.Command, msg: protocol._DecodedMsgType) -> None:
         """Handle the base protocol (P2P) messages."""
+        self.base_protocol = cast(P2PProtocol, self.base_protocol)
```

Comment:
Another cast I wouldn't think to be needed.

Reply:
solved, see previous comment